packages/async-flow/src/async-flow.js

```diff
@@ -0,0 +1,402 @@
+                  admin.done();
+                }
+              },
+            );
+          },
+          wake() {
+            const { state, facets } = this;
+            const { flow } = facets;
+
+            if (failures.has(flow)) {
+              throw failures.get(flow);
+            }
+            if (state.isDone) {
+              return;
+            }
+            if (tmp.for(flow).membrane) {
+              tmp.for(flow).membrane.wake();
+            } else {
+              flow.restart();
+            }
+          },
+          getOutcome() {
+            const { state, facets } = this;
+            const { outcomeKit } = state;
+            const { flow } = facets;
+
+            flow.wake();
+            return outcomeKit.vow;
+          },
+          dump() {
+            const { state } = this;
+            const { log } = state;
+
+            return log.dump();
+          },
+          getOptFatalProblem() {
+            const { facets } = this;
+            const { flow } = facets;
+
+            return failures.has(flow) ? failures.get(flow) : undefined;
+          },
+        },
+        admin: {
+          reset() {
+            const { state, facets } = this;
+            const { bijection, log } = state;
+            const { flow } = facets;
+
+            if (failures.has(flow)) {
+              failures.delete(flow);
+            }
+            if (eagerWakers.has(flow)) {
+              // For now, once an eagerWaker, always an eagerWaker
+              // eagerWakers.delete(flow);
+            }
+            if (tmp.for(flow).membrane) {
+              tmp.for(flow).membrane.stop();
+            }
+            tmp.resetFor(flow);
+            log.reset();
+            bijection.reset();
+
+            state.isDone = false;
+          },
+          done() {
+            const { state, facets } = this;
+            const { log } = state;
+            const { admin } = facets;
+
+            admin.reset();
+            state.isDone = true;
+            log.dispose();
+          },
+          panic(fatalProblem) {
+            const { state, facets } = this;
+            const { bijection, log } = state;
+            const { flow } = facets;
+
+            if (failures.has(flow)) {
+              failures.set(flow, fatalProblem);
+            } else {
+              failures.init(flow, fatalProblem);
+            }
+
+            if (tmp.for(flow).membrane) {
+              tmp.for(flow).membrane.stop();
+            }
+            tmp.resetFor(flow);
+            log.reset();
+            bijection.reset();
+
+            // This is a non-sensical return value, so arbitrary chaos
+            // may ensue from returning it. But at this point
+            // we should have successfully isolated this activation from
+            // having any observable effects on the host, aside from
+            // console logging and
+            // resource exhaustion, including infinite loops
+            return undefined;
```

Comment:
Can you make the return result describe this situation better?  Maybe something like:
```suggestion
            return `panic: see getFailures() for more information`;
```

Reply:
Even better: made it a `throw Fail`. Done.